### PR TITLE
Implement `getBufferType` for `flow.dispatch.tensor.load`

### PR DESCRIPTION
**Pull request by matthias-springer**
This function computes the future buffer type of a tensor and is required during bufferization to deduce a correct buffer type for block arguments (e.g., in SCF loops and unstructured control flow).

### CODE REVIEW

File: compiler/src/iree/compiler/Codegen/Interfaces/BufferizationInterfaces.cpp [hunk 1]
```diff
@@ -70,22 +70,23 @@ getMemrefTypeForTensor(IREE::Flow::DispatchTensorType tensorType,
...
       subspanOp.getResult().getType());
   assert(shapedType && shapedType.hasRank());
 
+  IRRewriter rewriter(subspanOp.getContext());
+  rewriter.setInsertionPoint(subspanOp);
+
   Value byteOffset = subspanOp.getByteOffset();
   MemRefLayoutAttrInterface layoutAttr = {};
   if (byteOffset && !matchPattern(byteOffset, m_Zero())) {
+    // TODO: Find a way to call convertByteOffsetToElementOffset without a
```
Comment (matthias-springer) [inline compiler/src/iree/compiler/Codegen/Interfaces/BufferizationInterfaces.cpp:88]: Do you see a way to address this?

Comment (MaheshRavishankar) [reply compiler/src/iree/compiler/Codegen/Interfaces/BufferizationInterfaces.cpp:88]: yeah you should be able to just get the type of the memref, leaving offset as dynamic. It should be possible to write a version of that method which just gets the type of the memref....

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]
Not sure what having an `IRRewriter` instantiated in the middle is going to do.

**Comment by benvanik** [comment]
Closing stale PR.

**Pull request closed without merging**